grapher_admin/views.py

```diff
@@ -1434,3 +1547,359 @@ def register_by_invite(request: HttpRequest, code: str):
+    allentities_list = []
+    for each in allentities:
+        allentities_list.append({'id': each['id'], 'name': each['name']})
+    allentities_list = sorted(allentities_list, key=lambda k: k['name'])
+    allentities_list = [item['id'] for item in allentities_list]
+    # we are splitting the entities to groups of 3 for querying
+    # this is still not very efficient, but better than querying values for each entity separately
+    # if we don't set this limitation and query for values from all entities, th db usually hangs -
+    # - when there are many rows present
+    entity_chunks = [allentities_list[x:x + 3] for x in range(0, len(allentities_list), 3)]
+
+    dataset_meta = {}
+
+    with connection.cursor() as source_cursor:
+        source_cursor.execute('select name, description from sources where id in (%s);' % ','.join([str(x) for x in source_ids]))
+        dataset_meta['sources'] = [{'name': item[0], 'description': item[1]} for item in source_cursor.fetchall()]
+
+    counter = 0
+    source_name_to_id = {}
+    for each in dataset_meta['sources']:
+        each['id'] = counter
+        source_name_to_id[each['name']] = counter
+        counter += 1
+
+    for each in vardata:
+        each['source_id'] = source_name_to_id[each['source_id']]
+
+    dataset_meta['variables'] = vardata
+
+    sql_query = 'SELECT `value`, `year`, data_values.`fk_var_id` as var_id, data_values.`fk_ent_id` as entity_id ' \
+                ' from data_values ' \
+                ' WHERE data_values.`fk_var_id` in (%s) AND ' \
+                'data_values.`fk_ent_id` in (%s) ORDER BY fk_ent_id, year, fk_var_id;'
+
+    metadata_filename = (unidecode(dataset.name) + '.json').replace('/', '_')
+    dataset_filename = (unidecode(dataset.name) + '.csv').replace('/', '_')
+
+    if old_dataset_name:
+        old_metadata_name = (unidecode(old_dataset_name) + '.json').replace('/', '_')
+        old_dataset_name = (unidecode(old_dataset_name) + '.csv').replace('/', '_')
+        old_metadata_file_path = os.path.join(current_dataset_folder, old_metadata_name)
+        old_dataset_file_path = os.path.join(current_dataset_folder, old_dataset_name)
+        old_metadata_file_path_escaped = shlex.quote(os.path.join(current_dataset_folder, old_metadata_name))
+        old_dataset_file_path_escaped = shlex.quote(os.path.join(current_dataset_folder, old_dataset_name))
+
+    metadata_file_path = os.path.join(current_dataset_folder, metadata_filename)
+    dataset_file_path = os.path.join(current_dataset_folder, dataset_filename)
+    metadata_file_path_escaped = shlex.quote(os.path.join(current_dataset_folder, metadata_filename))
+    dataset_file_path_escaped = shlex.quote(os.path.join(current_dataset_folder, dataset_filename))
+
+    delete_previous = False
+    if old_dataset_name is not None:
+        if os.path.isfile(old_dataset_file_path):
+            subprocess.check_output('git -C %s rm %s' %
```

Comment:
I think this can be simplified a bit by using `git mv`. I'd suggest batching the repo-modifying git operations into a single subprocess call at the end, to avoid leaving the repo in an intermediary state if errors occur. (you already mostly do this!)